server/src/services/library.service.ts

```diff
@@ -472,27 +470,35 @@ export class LibraryService extends BaseService {
     return JobStatus.SUCCESS;
   }
 
-  @OnJob({ name: JobName.LIBRARY_SYNC_ASSET, queue: QueueName.LIBRARY })
-  async handleSyncAsset(job: JobOf<JobName.LIBRARY_SYNC_ASSET>): Promise<JobStatus> {
-    const asset = await this.assetRepository.getById(job.id);
+  @OnJob({ name: JobName.LIBRARY_SYNC_ASSETS, queue: QueueName.LIBRARY })
+  async handleSyncAssets(job: JobOf<JobName.LIBRARY_SYNC_ASSETS>): Promise<JobStatus> {
+    for (const id of job.ids) {
+      await this.handleSyncAsset(id, job.importPaths, job.exclusionPatterns);
```

Comment:
Fetch all the assets first with `getByIds`. Group the ones to be marked offline, queued for metadata extraction, etc. as you check them, then do batched async calls at the end as needed. The try/catch for `stat` should still be scoped to the asset so one error won't torpedo the batch.

Reply:
Done